cpp/src/parquet/arrow/reader_internal.cc

```diff
@@ -747,8 +747,7 @@ Status TypedIntegralStatisticsAsScalars(const Statistics& statistics,
       using CType = typename StatisticsType::T;
       return MakeMinMaxScalar<CType, StatisticsType>(statistics, min, max);
     default:
-      return Status::NotImplemented("Cannot extract statistics for type ",
-                                    logical_type->ToString());
+      return Status::NotImplemented("Cannot extract statistics for type ");
```

Comment:
It turns out that this one gave a non-negligible performance boost because there's a lot of type we don't support, multiply by the number of row group and columns...

Reply:
Maybe this should return `optional<pair<shared_ptr<Scalar>, shared_ptr<Scalar>>>` instead so that callers can decide whether a verbose error is appropriate? Follow up, in any case

Reply:
KISS, we should just extend MakeMinMaxScalar to support the other types.